xlators/mgmt/glusterd/src/glusterd-op-sm.c

```diff
@@ -2517,27 +2517,19 @@ glusterd_set_brick_mx_opts(dict_t *dict, char *key, char *value,
 
 static int
 glusterd_set_brick_graceful_cleanup(dict_t *dict, char *key, char *value,
-                                    char **op_errstr)
+                                    glusterd_conf_t *priv)
 {
-    int32_t ret = -1;
-    xlator_t *this = NULL;
-    glusterd_conf_t *priv = NULL;
-
-    this = THIS;
-    GF_VALIDATE_OR_GOTO("glusterd", this, out);
-    GF_VALIDATE_OR_GOTO(this->name, dict, out);
-    GF_VALIDATE_OR_GOTO(this->name, key, out);
-    GF_VALIDATE_OR_GOTO(this->name, value, out);
-    GF_VALIDATE_OR_GOTO(this->name, op_errstr, out);
-
-    ret = 0;
-
-    priv = this->private;
+    int ret = 0;
+    char *dup_value = NULL;
 
-    if (!strcmp(key, GLUSTERD_BRICK_GRACEFUL_CLEANUP)) {
-        ret = dict_set_dynstrn(priv->opts, GLUSTERD_BRICK_GRACEFUL_CLEANUP,
-                               SLEN(GLUSTERD_BRICK_GRACEFUL_CLEANUP),
-                               gf_strdup(value));
+    if (!strcmp(key, GLUSTER_BRICK_GRACEFUL_CLEANUP)) {
+        dup_value = gf_strdup(value);
+        if (!dup_value) {
+            ret = -1;
+            goto out;
+        }
+        ret = dict_set_dynstrn(priv->opts, GLUSTER_BRICK_GRACEFUL_CLEANUP,
```

Comment:
What happens with `dup_value` if this fails ?

Reply:
glusterd clit would not be able to set a key